### PR TITLE
Add error hook

### PR DESCRIPTION
**Pull request by Kelerchian**

### CODE REVIEW

File: machine-runner/README.md [hunk 1]
```diff
@@ -299,6 +344,23 @@ A zod-enabled event factory will have these additional features enabled:
   })
   ```
 
+### Global Event Emitter
+
+Some global event emitters are provided.
+This event emitters will emit events from all machine runners in the same process.
```
Comment (jmg-duarte) [inline machine-runner/README.md:350]:
```suggestion
These event emitters will emit events from all machine runners in the same process.
```

File: machine-runner/README.md [hunk 2]
```diff
@@ -234,6 +234,51 @@ If a robot saw itself winning, started the mission, and then discovers that its
...
+    // ...
+  }
+})
+
+await untilWareHouseIsNotUsedAnymore()
+
+warehouse.destroy()
+```
+
+`error` event can be used to capture errors that happens from machine-runner.
```
Comment (jmg-duarte) [inline machine-runner/README.md:258]:
```suggestion
`error` event can be used to capture errors from machine-runner.
```

File: machine-runner/README.md [hunk 3]
```diff
@@ -234,6 +234,51 @@ If a robot saw itself winning, started the mission, and then discovers that its
 
 ## Features
 
+### Observe Changes and Errors
+
+An alternative use case of a machine runner is to listen to its events.
+
+`next` event emits states whenever a new state is calculated.
```
Comment (jmg-duarte) [inline machine-runner/README.md:241]:
```suggestion
The `next` event emits states whenever a new state is calculated.
```

File: machine-runner/README.md [hunk 4]
```diff
@@ -234,6 +234,51 @@ If a robot saw itself winning, started the mission, and then discovers that its
...
+Calling `destroy` is required when the machine is not used to sever subscription from Actyx.
```
Comment (jmg-duarte) [inline machine-runner/README.md:242]:
```suggestion
When not using the machine, calling `destroy` is required to close the connection to Actyx.
```

**Review by rkuhn — APPROVED** [review]
Yes, looks good to me apart from José’s remarks.

**Pull request merged**